internal/resourcehelper/helper.go

```diff
@@ -0,0 +1,251 @@
+package resourcehelper
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+//go:generate mockgen -source=helper.go -package=resourcehelper -destination=mock_helper_api.go
+
+type Helper interface {
+	IsNamespaced(kind string) bool
+	IsNotUpdateable(kind string) bool
+	NeedsResourceVersionUpdate(kind string) bool
+	UpdateResourceVersion(req *unstructured.Unstructured, found *unstructured.Unstructured) error
+	SetNodeSelectorTerms(obj *unstructured.Unstructured, terms map[string]string) error
+	IsOneTimer(obj *unstructured.Unstructured) (bool, error)
+	SetLabel(obj *unstructured.Unstructured, label string) error
+	SetMetaData(obj *unstructured.Unstructured, nm string, ns string)
+}
+
+func New() Helper {
+	return &resourceHelper{}
+}
+
+type resourceHelper struct{}
+
+func (rh resourceHelper) IsNamespaced(kind string) bool {
+	if kind == "Namespace" ||
+		kind == "ClusterRole" ||
+		kind == "ClusterRoleBinding" ||
+		kind == "SecurityContextConstraint" ||
+		kind == "SpecialResource" {
+		return false
+	}
+	return true
+}
+
+func (rh resourceHelper) IsNotUpdateable(kind string) bool {
+	// ServiceAccounts cannot be updated, maybe delete and create?
+	if kind == "ServiceAccount" || kind == "Pod" {
+		return true
+	}
+	return false
+}
+
+// Some resources need an updated resourceversion, during updates
+func (rh resourceHelper) NeedsResourceVersionUpdate(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "Build" ||
+		kind == "BuildRun" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" ||
+		kind == "CSIDriver" ||
+		kind == "Issuer" ||
+		kind == "CustomResourceDefinition" ||
+		kind == "Certificate" ||
+		kind == "SpecialResource" ||
+		kind == "OperatorGroup" ||
+		kind == "CertManager" ||
+		kind == "MutatingWebhookConfiguration" ||
+		kind == "ValidatingWebhookConfiguration" ||
+		kind == "Deployment" ||
+		kind == "ImagePolicy" {
+		return true
+	}
+	return false
+
```

Comment:
```suggestion
	return kind == "SecurityContextConstraints" ||
		kind == "Service" ||
		kind == "ServiceMonitor" ||
		kind == "Route" ||
		kind == "Build" ||
		kind == "BuildRun" ||
		kind == "BuildConfig" ||
		kind == "ImageStream" ||
		kind == "PrometheusRule" ||
		kind == "CSIDriver" ||
		kind == "Issuer" ||
		kind == "CustomResourceDefinition" ||
		kind == "Certificate" ||
		kind == "SpecialResource" ||
		kind == "OperatorGroup" ||
		kind == "CertManager" ||
		kind == "MutatingWebhookConfiguration" ||
		kind == "ValidatingWebhookConfiguration" ||
		kind == "Deployment" ||
		kind == "ImagePolicy"
```

Reply:
What would you say about changing `if` to a `switch`? For example:
```
switch (kind) {
    case SecurityContextConstraints",
        "Service",
        "ServiceMonitor",
        "Route",
        ....
        "Build":
        return true
    default:
        return false
}
``

Reply:
Straight `return` with the condition looks simpler to me, but I don't have a strong opinion here.

Reply:
PTAL, I changed `IsNamespaced()`, `IsNotUpdateable()`, `NeedsResourceVersionUpdate()`, `SetNodeSelectorTerms()`, `setSubResourceLabel()`. I personally find such `switch`es more clear than `if` with a lot of `kind ==` duplicated.

Reply:
Actually, if we want to do it nicely, how about a new `StringSet` type that would wrap `map[string]bool`? Then those sets could be initialized only once outside of the methods.

Reply:
Well... I'm not convinced that Set would look better. It would live outside a function (and only used in that single function). Now it's self-contained (i.e. function+data for decision). I wouldn't want to have a Set for every function that does this kind of logic (I mentioned them in previous comment).
If we're talking about performance (in regards to single init), wouldn't switch be optimized compile-time so no runtime init?